File: apps/ios/TwoDo/Models/Couple.swift
import Foundation

struct Couple: Codable, Identifiable {
    let id: String
    var name: String
    var inviteCode: String?
    var inviteCodeExpiresAt: Date?
    let createdAt: Date
    let updatedAt: Date
}

struct CoupleWithMembers: Codable {
    let couple: Couple
    let members: [User]
}

// MARK: - Couple Request/Response Models

struct CreateCoupleRequest: Codable {
    let name: String
}

struct CreateCoupleResponse: Codable {
    let couple: Couple
}

struct JoinCoupleRequest: Codable {
    let inviteCode: String
}

struct JoinCoupleResponse: Codable {
    let couple: Couple
}

struct GenerateInviteResponse: Codable {
    let inviteCode: String
    let expiresAt: Date
}

struct GetCoupleResponse: Codable {
    let couple: Couple
    let members: [User]
}

struct UpdateCoupleRequest: Codable {
    let name: String
}

struct UpdateCoupleResponse: Codable {
    let couple: Couple
}

File: packages/shared/src/types/couple.ts
export interface Couple {
  id: string;
  name: string;
  inviteCode?: string;
  inviteCodeExpiresAt?: Date;
  createdById: string;
  createdAt: Date;
  updatedAt: Date;
}

export type CoupleRole = 'owner' | 'member';

export interface CoupleUser {
  coupleId: string;
  userId: string;
  role: CoupleRole;
  joinedAt: Date;
}

export interface CoupleWithMembers extends Couple {
  members: Array<{
    userId: string;
    displayName: string;
    avatarUrl?: string;
    role: CoupleRole;
  }>;
}
